### PR TITLE
Add hasIndex() to OCP\FullTextSearch\Model\IndexDocument

### PR DESCRIPTION
**Pull request by ArtificialOwl**

### CODE REVIEW

File: lib/public/FullTextSearch/Model/IndexDocument.php [hunk 1]
```diff
@@ -183,6 +183,17 @@ public function __construct(string $providerId, string $documentId) {
...
 	}
 
+	/**
+	 * return if Index is defined.
+	 *
+	 * @since 16.0.0
+	 *
+	 * @return bool
+	 */
+	final public function gotIndex(): bool {
```
Comment (go2sh) [inline lib/public/FullTextSearch/Model/IndexDocument.php]: hasIndex feels more usual. :-)

Comment (nickvergessen) [reply lib/public/FullTextSearch/Model/IndexDocument.php]: yes,

Also if you avoid usage of final, Mocks can be used in unit tests

Comment (daita) [reply lib/public/FullTextSearch/Model/IndexDocument.php]: final was asked by @MorrisJobke if I remember well.

Also switched to hasIndex()

**Review by rullzer — APPROVED** [review]

File: lib/public/FullTextSearch/Model/IndexDocument.php [hunk 2]
```diff
@@ -183,6 +183,17 @@ public function __construct(string $providerId, string $documentId) {
...
 
+	/**
+	 * return if Index is defined.
+	 *
+	 * @since 16.0.0
+	 *
+	 * @return bool
+	 */
+	final public function hasIndex(): bool {
+		return !($this->index === null);
```
Comment (nickvergessen) [inline lib/public/FullTextSearch/Model/IndexDocument.php]:
```suggestion
		return $this->index !== null;
```

**Review by nickvergessen — APPROVED** [review]

**Comment by MorrisJobke** [comment]
Let me rebase to trigger CI.

**Pull request merged**